compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/types/FunctionTypeTest.java

```diff
@@ -72,4 +72,9 @@ public void testRecoveryInFuncTypeWithReturns() {
     public void testRecoveryInFuncTypeUsagesInTypeDescContexts() {
         testFile("func-type/func_type_source_07.bal", "func-type/func_type_assert_07.json");
     }
+
+    @Test
+    public void testRecoveryInFuncTypeWithoutSignature() {
+        testFile("func-type/func_type_source_07.bal", "func-type/func_type_assert_07.json");
```

Comment:
```suggestion
        testFile("func-type/func_type_source_12.bal", "func-type/func_type_assert_12.json");
```

Let's add more recovery cases eg
```
function a myfunc; //missing equal
function foo ) {
} // missing open parenthesis
function b % = func2; // additional tokens
```
Test cases to test recovery of relative resource path

Reply:
Added.